Using AsyncBufReadExt with a BufStream

<!--
Thank you for reporting an issue.

Please fill in as much of the template below as you're able.
-->

## Version
rustc 1.40.0 (73528e339 2019-12-16)

lazy_static = "1.4.0"
ordered-float = "1.0.2"
rand = "0.7.2"
serde = "1.0.60"
serde_json = "1.0.44"
simd-json = "0.2.2"
tokio = { version = "0.2.6", features = ["rt-threaded", "sync", "time", "fs", "macros", "io-util", "tcp", "dns"] }
tokio-tls = "0.3.0"
native-tls = "0.2.3"
crossbeam = "0.7.3"
chrono = { version = "0.4.10", features = ["serde"] }
async-trait = "0.1.22"
reqwest = { version = "0.10.0", features = ["default-tls", "gzip"] }
<!--
List the versions of all `tokio` crates you are using. The easiest way to get
this information is using `cargo-tree`.

`cargo install cargo-tree`
(see install here: https://github.com/sfackler/cargo-tree)

Then:

`cargo tree | grep tokio`
-->

## Platform
Ubuntu 18.04.2 LTS
<!---
Output of `uname -a` (UNIX), or version and 32 or 64-bit (Windows)
-->

## Description

I am trying to use an [AsyncBufReadExt](https://docs.rs/tokio/0.2.6/tokio/io/trait.AsyncBufReadExt.html) function to read data from a [BufStream](https://docs.rs/tokio/0.2.6/tokio/io/struct.BufStream.html) but I am receiving the following error:

```
error[E0599]: no method named `read_line` found for type `tokio::io::util::buf_stream::BufStream<tokio_tls::TlsStream<tokio::net::tcp::stream::TcpStream>>` in the current scope
  --> src/bin/main.rs:41:29
   |
41 |     let result = buf_stream.read_line(&mut line).await;
   |                             ^^^^^^^^^ method not found in `tokio::io::util::buf_stream::BufStream<tokio_tls::TlsStream<tokio::net::tcp::stream::TcpStream>>`
   |
   = note: the method `read_line` exists but the following trait bounds were not satisfied:
           `tokio::io::util::buf_stream::BufStream<tokio_tls::TlsStream<tokio::net::tcp::stream::TcpStream>> : tokio::io::util::async_buf_read_ext::AsyncBufReadExt`
```

The code:
```
use native_tls::TlsConnector;
use tokio::net::TcpStream;
use tokio::io::{BufStream, BufReader, AsyncBufReadExt};
use tokio_tls::{TlsConnector as TokioTlsConnector, TlsStream};

async fn doesnt_compile() {
    let connector = TlsConnector::new().unwrap();
    let connector = TokioTlsConnector::from(connector);
    let tcp_stream = TcpStream::connect("example.com:443").await.unwrap();
    let tls_stream = connector.connect("example.com", tcp_stream).await.unwrap();
    let buf_stream = BufStream::new(tls_stream);
    
    let mut line = String::new();
    let result = buf_stream.read_line(&mut line).await;
}
```

If I replace BufStream by a BufReader, the following code compiles:
```
async fn compiles() {
    let connector = TlsConnector::new().unwrap();
    let connector = TokioTlsConnector::from(connector);
    let tcp_stream = TcpStream::connect("example.com:443").await.unwrap();
    let tls_stream = connector.connect("example.com", tcp_stream).await.unwrap();
    let mut buf_reader = BufReader::new(tls_stream);
    
    let mut line = String::new();
    let result = buf_reader.read_line(&mut line).await;
}
```

BufStream and BufReader both implement the AsyncBufRead trait according to https://docs.rs/tokio/0.2.6/tokio/io/struct.BufStream.html and https://docs.rs/tokio/0.2.6/tokio/io/struct.BufReader.html so why can't I use the AsyncBufReadExt functions with a BufStream?

Thanks

<!--

Enter your issue details below this comment.

One way to structure the description:

<short summary of the bug>

I tried this code:

<code sample that causes the bug>

I expected to see this happen: <explanation>

Instead, this happened: <explanation>
-->

--- a/tokio/src/io/util/buf_stream.rs
+++ b/tokio/src/io/util/buf_stream.rs
@@ -131,7 +131,7 @@ impl<RW: AsyncRead + AsyncWrite> AsyncRead for BufStream<RW> {
     }
 }
 
-impl<RW: AsyncBufRead + AsyncRead + AsyncWrite> AsyncBufRead for BufStream<RW> {
+impl<RW: AsyncRead + AsyncWrite> AsyncBufRead for BufStream<RW> {
     fn poll_fill_buf(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<io::Result<&[u8]>> {
         self.project().inner.poll_fill_buf(cx)
     }

